data_structures/src/chain.rs

```diff
@@ -179,10 +179,9 @@ pub struct BlockInChain {
     pub data_request_pool: DataRequestPool,
 }
 
-/// Any reference to a Hashable type is also Hashable
-impl<'a, T: Hashable> Hashable for &'a T {
+impl<T: AsRef<[u8]>> Hashable for T {
```

Comment:
This may break the former property "_any reference to a Hashable type is also Hashable_".

Reply:
Yes, I thought about it, but the case for "_any ref to a Hashable is also Hashable_" is needed only if we define a generic function like:
```rust
pub fn do_something<T: Hashable>(x: T) {
  // ...
}
```
and we want to allow `T` to be either a Hashable or a ref to a Hashable. After some thought I think this _feature_ won't be really needed and the more natural approach if we don't care about the ownership is to just ask for a ref to a Hashable:
```rust
pub fn do_something<T: Hashable>(x: &T) {
  // ...
}
```
in fact this is the case for the `sign` function in the signature manager:
```rust
pub fn sign<T>(data: &T) -> impl Future
where
    T: Hashable,
{
  // ...
}
```

Unfortunately I cannot leave both impls because they clash.